src/plugins/console/server/config.ts

```diff
@@ -83,8 +83,8 @@ const config7x: PluginConfigDescriptor<ConsoleConfig7x> = {
     ui: true,
   },
   schema: schema7x,
-  deprecations: ({ deprecate, unused }) => [
-    unused('ssl'),
+  deprecations: ({ unused }) => [
+    unused('ssl', { level: 'warning' }),
```

Comment:
Yes this should be critical as setting `console.ssl` will throw an error when starting Kibana.  
I have been looking for old documentation about the `console.ssl` setting and haven't found it. I see that this deprecation has been introduced in `7.7.` (https://github.com/elastic/kibana/blob/7.7/src/plugins/console/server/index.ts) and I am not entirely sure why.

Reply:
> as setting console.ssl will throw an error when starting Kibana

It will not, as this very deprecation removes it from the config's data.

`critical` is supposed to be for deprecations that will break when not fixed on the next release, so for config deprecations that's basically deprecations that we're stop supporting. Are we expecting to get rid of this `unused('ssl')` deprecation in the short/mid term?

Reply:
(but I'm very fine changing the level if we want to)

Reply:
I did some investigation and I think `console.ssl` has been deprecated for many major versions.

Based on [this line](https://github.com/elastic/kibana/pull/55690/files#diff-c16cb3e70215291743f8922742abafc8450a411d3289bd90761b5c394add859eL124), it looks like there used to a setting for `sense.ssl`:

```js
if (options.ssl && options.ssl.verify) {
  throw new Error('sense.ssl.verify is no longer supported.');
}
```

For historical context, Console used to be a standalone app called Sense. In Kibana 5, it was renamed Console and incorporated into Kibana.

I did some Googling and found https://github.com/elastic/kibana/issues/7128, which references some legacy Sense documentation which we happen to still be hosting: https://www.elastic.co/guide/en/sense/current/installing.html#configuration.

What's interesting about these docs is that they make no mention of this `sense.ssl` setting (it was moved under `sense.proxyConfig.ssl`). Based on this information, my inference is that this setting has been deprecated since 5.0, but we just never formally communicated that. In that case the resolution should be for 7.17 to communicate that support for this setting will be removed in 8.x, and then reject its use in 8.x. Does that sound correct @sebelga and @pgayvallet?

Reply:
> In that case the resolution should be for 7.17 to communicate that support for this setting will be removed in 8.x, and then reject its use in 8.x. Does that sound correct

That seems correct to me

Reply:
It seems correct. But going one step further: how would it even be possible for a user to have `console.ssh` setting in their `kibana.yml` file if our config schema does not allow it? Kibana does not start if you declare it. So why do we add a deprecation for something that can't be done in the first place?

Reply:
`unused('ssl')` effectively remove the entry from the config, meaning that the block @cjcenizal linked in https://github.com/elastic/kibana/pull/123411#discussion_r789254946 is effectively never entered

Reply:
@pgayvallet Thanks for the explanation. I'm having trouble finding docs for `unused()` -- can you point me to them? It sounds like you're saying that it ignores the presence of the setting in `kibana.yml`, allowing Kibana to start up as expected. I'm confused by that behavior because I would want Kibana to fail to start if an unexpected setting is in use. I'm also confused about what the second argument does (`'warning'`).

Reply:
@cjcenizal I changed the deprecation level to `critical` in https://github.com/elastic/kibana/pull/123411/commits/edd24cccad84f0a56f85c06c6258f380df4e96c5. Are you fine merging this PR first and then letting the @elastic/platform-deployment-management team doing the proper cleanup in 7.17/8/8.1 if you want to totally remove this deprecation in 8+?

Reply:
@pgayvalle I don't want to block you but I'm still really confused by how these changes will manifest in 8.1. 😄  When you say you changed it to `critical`, what is the impact of that on how Kibana behaves?

Reply:
> When you say you changed it to critical, what is the impact of that on how Kibana behaves?

As the default level for deprecations was `critical` before this PR, it does nothing apart from explicitly setting this level instead of it being inferred (also the deprecation's level has no real impact outside the being surfaced in UA).